src/app/pages/details/common/get-this-title-files/give-before-pdf/thank-you-form.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import {useUserModel} from '~/models/usermodel';
+import './thank-you-form.scss';
+
+function FormWithAfterSubmit({
+    afterSubmit,
+    responseId='form-response',
+    children,
+    onSubmit: firstOnSubmit,
+    ...formProps
+}) {
+    const iframeRef = React.useRef();
+
+    function onSubmit(event) {
+        if (firstOnSubmit) {
+            firstOnSubmit(event);
+            if (event.defaultPrevented) {
+                return;
+            }
+        }
+        const iframe = iframeRef.current;
+        const runAfterSubmit = () => {
+            afterSubmit();
+            iframe.removeEventListener('load', runAfterSubmit);
+        };
+
+        iframe.addEventListener('load', runAfterSubmit);
+    }
+
+    return (
+        <React.Fragment>
+            <iframe
+                name={responseId} id={responseId} className="hidden"
+                src="" width="0" height="0" tabIndex="-1"
+                ref={iframeRef}
+            />
+            <form target={responseId} {...formProps} onSubmit={onSubmit}>
+                {children}
+            </form>
+        </React.Fragment>
+    );
+}
+
+function ThankYou({userModel={}, link, close}) {
+    const first = userModel.first_name;
+    const last = userModel.last_name;
+    const school = userModel.self_reported_school;
+
+    function afterSubmit() {
+        window.open(link);
+        close();
+    }
```

Comment:
sneaky stuff here.  Are you sure the the window.open will be actually open, since it's opening inside an invisible iframe?  Sure seems like that's exactly what content blockers will try to prevent

Reply:
The `window` object is still the global context. It doesn't change to refer to the iframe just because an iframe event is calling the function. What's going on is that the form submission response loads in the iframe; when that happens, then the new window opens displaying the PDF.